### ISSUE TITLE
Source files partially copied when transient agent is destroyed (i.e. Docker) 

### ISSUE THREAD

**Issue by capadgett** [jenkinsci/code-coverage-api-plugin#82]
Thanks for the awesome plugin (efficient, snappy interface, etc.).

When using a transient docker agent, I see files in the build directory on the Jenkins master that are partially copied or empty, and have the trailing `_copied` at the end of the file name.

I am not positive, but I think this issue is related to not waiting for the ExecutorService in `io/jenkins/plugins/coverage/source/DefaultSourceFileResolver.java`:
```
   81         ExecutorService service = Executors.newFixedThreadPool(5);
   82
   83         paints.forEach((sourceFilePath, paint) -> {
   84             service.submit(() -> {                                              
```

I should also mention that I am using the `publishCoverage` pipeline step in a `stage{ post{} }`, not a `pipeline{ post{} }`, but I wouldn't think that should matter.

Might be related to fix for #77

**Comment by cizezsy** [comment]
@capadgett You are right, ExecutorService cannot prevent agent from destroyed so not be able to copy all source files. Thanks for the feedback, I will try to fix it in next release.

**Comment by cizezsy** [comment]
@capadgett Fixed in version 1.0.11, thanks :P
